pyccl/halos/concentration.py

```diff
@@ -353,6 +354,169 @@ def _concentration(self, cosmo, M, a):
+                    self.b0 = 1.83
+                    self.b1 = 3.52
+                    self.c_alpha = -0.18
+                else:  # now it's 200c
+                    self.kappa = 1.10
+                    self.a0 = 2.30
+                    self.a1 = 1.64
+                    self.b0 = 1.72
+                    self.b1 = 3.60
+                    self.c_alpha = 0.32
+        else:  # use profile fitting method
+            if self.relaxed:  # fit only relaxed halos
+                if self.mdef.Delta == 'vir':
+                    self.kappa = 1.22
+                    self.a0 = 2.52
+                    self.a1 = 1.87
+                    self.b0 = 2.13
+                    self.b1 = 4.19
+                    self.c_alpha = -0.017
+                else:  # now it's either 200c or 500c
+                    if int(self.mdef.Delta) == 200:
+                        self.kappa = 0.60
+                        self.a0 = 2.14
+                        self.a1 = 2.63
+                        self.b0 = 1.69
+                        self.b1 = 6.36
+                        self.c_alpha = 0.37
+                    else:  # now it's 500c
+                        self.kappa = 0.38
+                        self.a0 = 1.44
+                        self.a1 = 3.41
+                        self.b0 = 2.86
+                        self.b1 = 2.99
+                        self.c_alpha = 0.42
+            else:  # fit all halos
+                if self.mdef.Delta == 'vir':
+                    self.kappa = 1.64
+                    self.a0 = 2.67
+                    self.a1 = 1.23
+                    self.b0 = 3.92
+                    self.b1 = 1.30
+                    self.c_alpha = -0.19
+                else:  # now it's either 200c or 500c
+                    if int(self.mdef.Delta) == 200:
+                        self.kappa = 1.19
+                        self.a0 = 2.54
+                        self.a1 = 1.33
+                        self.b0 = 4.04
+                        self.b1 = 1.21
+                        self.c_alpha = 0.22
+                    else:  # now it's 500c
+                        self.kappa = 1.83
+                        self.a0 = 1.95
+                        self.a1 = 1.17
+                        self.b0 = 3.57
+                        self.b1 = 0.91
+                        self.c_alpha = 0.26
+
+    def _dlsigmaR(self, cosmo, M, a):
+        # kappa multiplies radius, so in log, 3*kappa multiplies mass
+        logM = 3*np.log10(self.kappa) + np.log10(M)
+
+        status = 0
+        dlns_dlogM, status = lib.dlnsigM_dlogM_vec(cosmo.cosmo, a, logM,
+                                                   len(logM), status)
+        check(status)
+        return -3/np.log(10) * dlns_dlogM
+
+    def _G(self, x, n_eff):
+        fx = np.log(1 + x) - x / (1 + x)
+        G = x / fx**((5 + n_eff) / 6)
+        return G
+
+    def _G_inv(self, Arg, n_eff):
+        from scipy.optimize import root_scalar
```

Comment:
Would it be worth importing this at the top of the file?

Reply:
Minimal overhead either way, but I moved it.